packages/data/lib/di/injector_data.dart

```diff
@@ -17,32 +22,65 @@ void initInjectorData() {
 
 void _initModuleApi() {
   GetIt.instance.registerSingleton<Dio>(
-    _buildMovieDio(
-      GetIt.instance.get<HeaderInterceptor>(),
+    _buildMovieDioTraktApi(
+      GetIt.instance.get<HeaderInterceptorTraktApi>(),
     ),
+    instanceName: traktApi,
   );
+
+  GetIt.instance.registerSingleton<ApiBaseService<ServicePayLoad>>(
+    ApiBaseServiceImpl(
+      GetIt.instance.get(
+        instanceName: traktApi,
+      ),
+    ),
+    instanceName: traktService,
+  );
+
+  GetIt.instance.registerSingleton<Dio>(
```

Comment:
- We use Dio right away when creating the application, so we will have to wait longer from the factory registration, since it will be called when loading movies;
- in case it really requires a long wait, we have LazySingleton;

Reply:
can u advice where do you read that factory creating instances slower than singleton? 
from doc: 
Only the first time you call get<T>() this factory function will be called to create a new instance. After that you will always get the same instance returned.
It also creates instance after first call, like factory. there no any time benefits.
Also, your Dio Instance will be "leave" inside your repository instance which is Singleton (and u can make factory of this dependency as lazySinglton). So, you call to create Dio only one time, why did you need it as Singleton?

Reply:
@sergei-mikhailovskii-idf